zio-http/jvm/src/main/scala/zio/http/netty/WebSocketAppHandler.scala

```diff
@@ -47,13 +47,14 @@ private[zio] final class WebSocketAppHandler(
     event: ChannelEvent[JWebSocketFrame],
     close: Boolean = false,
   ): Unit = {
-    zExec.runUninterruptible(ctx, NettyRuntime.noopEnsuring)(
-      queue.offer(event.map(frameFromNetty)) *>
-        (onComplete match {
-          case Some(promise) if close => promise.succeed(ChannelState.Invalid)
-          case _                      => ZIO.unit
-        }),
-    )
+    // IMPORTANT: Offering to the queue must be run synchronously to avoid messages being added in the wrong order
+    // Since the queue is unbounded, this will not block the event loop
+    // TODO: We need to come up with a design that doesn't involve running an effect to offer to the queue
+    zExec.unsafeRunSync(queue.offer(event.map(frameFromNetty)))
+    onComplete match {
+      case Some(promise) if close => promise.unsafe.done(Exit.succeed(ChannelState.Invalid))
+      case _                      => ()
+    }
```

Comment:
This is the most important change, we need to ensure that messages are offered to the queue synchronously to avoid them being added in the wrong order